superset/security/api.py

```diff
@@ -158,3 +164,64 @@ def guest_token(self) -> Response:
             return self.response_400(message=error.message)
         except ValidationError as error:
             return self.response_400(message=error.messages)
+
+    @expose("/users/related/<int:pk>", methods=("DELETE",))
+    @protect()
+    @safe
+    @statsd_metrics
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}" f".delete",
+        log_to_statsd=False,
+    )
+    def delete(self, pk: int) -> Response:
+        """Delete an User and all related records.
+        ---
+        delete:
+          summary: Delete an user and all related records
+          parameters:
+          - in: path
+            schema:
+              type: integer
+            name: pk
+          responses:
+            200:
+              description: User deleted
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      message:
+                        type: string
+            401:
+              $ref: '#/components/responses/401'
+            403:
+              $ref: '#/components/responses/403'
+            404:
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        try:
+            user = db.session.query(security_manager.user_model).get(pk)
+            if not user:
+                raise Exception("User not found")
+            db.session.query(FavStar).filter_by(user_id=pk).delete()
+            db.session.query(Log).filter_by(user_id=pk).delete()
+            db.session.query(KeyValueEntry).filter(KeyValueEntry.created_by_fk==pk).delete()
+            db.session.query(Slice).filter(Slice.created_by_fk==pk).delete()
```

Comment:
Due to the foreign key constraints and non-cascading logic one simply cannot delete a table, database, etc. as there likely will be other charts, datasets, etc. which rely on these. This is a safeguard to ensure one doesn't wipe out vast swaths of data by mistake.